### PR TITLE
Update Menu to v3.0 to make column consistent even with authentication

### PR DESCRIPTION
**Pull request by Ricokola**
Related to: WET-248

- Adjust column from 12 to 8 to the Header where the GCWeb Menu is to be consistent whether there is an Authentication section or not.
- Adjust and enhance the Header documentation to reflect changes and to update to version 3.0.
- Enhancement of the GCWeb documentation (Content change only).
- Update `col-xs-offset-7 col-xs-5` by `col-xs-offset-6 col-xs-6`

### CODE REVIEW

**Review by duboisp — CHANGES_REQUESTED** [review]
The home page template still use the `col-sm-12`, please update that working example

**Comment by Ricokola** [comment]
@duboisp ready to be reviewed

File: sites/header/deprecated/home-v2.html [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+			<hr>
+			<div class="container">
+				<div class="row">
+					{%- unless page.nomenu -%}
+						{%- if page.auth -%}
+							<div class="col-md-8">
+								{% include gcweb-menu/sitemenu.html %}
+							</div>
+						{%- else -%}
+							<div class="col-md-12">
```
[the remark this answers is not in the document]
Comment (duboisp) [reply sites/header/deprecated/home-v2.html:47]: This layout need to be updated. That is where I found the reference to col-sm-12. https://github.com/wet-boew/GCWeb/blob/master/templates/home/layouts/home.html#L42

**Comment by Ricokola** [comment]
@duboisp sites/header/deprecated/home-v2.html has been updated as suggested and the PR is now ready for review.

**Review by duboisp — CHANGES_REQUESTED** [review]
This file need to be updated too: https://github.com/wet-boew/GCWeb/blob/master/templates/home/layouts/home.html

+ Add some little migration instruction

**Comment by duboisp** [comment]
* [ ] Please add a small description of the change that happen on the home page template in the home page documentation page
* [ ] For the home page template, I do think we can directly use the header include instead of duplicating the logic.
* [ ] Add a section in the home page documentation where it list all production instance of the home page. Basically the EN and FR page on Canada.ca

When that is completed, we will be good to merge

File: sites/header/header-docs-en.html [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+	</tr>
+	<tr>
+		<td>n.a.</td>
+		<td>Version 3.0</td>
+		<td>Version 1.0</td>
+		<td>n.a.</td>
+	</tr>
+</table>
+
+<h2>Before and After</h2>
```
Comment (GormFrank) [inline sites/header/header-docs-en.html:59]: Je mettrais le titre "Before and After" en tant que H3 en-dessous d'un nouveau H2 nommé "Template". De ce fait, le titre "Previous versions" doit aussi devenir un H3 toujours sous le "Template", puis "Working examples" deviendra un H4. (Faire de même pour le Français)

File: templates/home/api-fr.html [hunk 3]
```diff
@@ -44,6 +49,31 @@ <h3>Component list and latest version</h3>
 	<li>Government initiatives - Version 1.0</li>
 </ul>
 
+<h3 id="home-page-banner">Home page header</h3>
```
Comment (GormFrank) [inline templates/home/api-fr.html:52]: typo
```suggestion
<h3 id="home-page-header">Home page header</h3>
```

File: templates/home/api-fr.html [hunk 4]
```diff
@@ -22,19 +22,24 @@
...
+	<dt>Availble instances</dt>
+	<dd>https://www.canada.ca/en.html</dd>
+	<dd>https://www.canada.ca/fr.html</dd>
 </dl>
 
 <h2>Technical specification</h2>
 
 <h3>Component list and latest version</h3>
 <ul>
+	<li><a href="#home-page-header">Home page header</a> - Version 2.0</li>
```
Comment (GormFrank) [inline templates/home/api-fr.html:42]: typo
```suggestion
	<li><a href="#home-page-header">Home page header</a> - Version 3.0</li>
```

File: templates/home/api-en.html [hunk 5]
```diff
@@ -44,6 +49,30 @@ <h3>Component list and latest version</h3>
 	<li>Government initiatives - Version 1.0</li>
 </ul>
 
+<h3 id="home-page-banner">Home page header</h3>
```
Comment (GormFrank) [inline templates/home/api-en.html:52]: typo
```suggestion
<h3 id="home-page-header">Home page header</h3>
```

File: templates/home/api-en.html [hunk 6]
```diff
@@ -22,19 +22,24 @@
...
+	<dd>2022-06-29</dd>
 
 	<dt>Conforming to</dt>
 	<dd>Going to be: Content and IA spec 2.0</dd>
 
 	<dt>Specification</dt>
 	<dd><a href="https://design.canada.ca/mandatory-templates/home-page.html">Home page for Canada.ca</a>
+
+	<dt>Availble instances</dt>
+	<dd>https://www.canada.ca/en.html</dd>
```
Comment (GormFrank) [inline templates/home/api-en.html:34]: Should be links <a> and in the same <dd> element (could be separated by a comma "," or the word "and").

File: templates/home/layouts/home.html [hunk 7]
```diff
@@ -8,55 +8,22 @@
...
+		<meta charset="utf-8">
+		{% include license.html %}
+		<title>{{ page-title }} - {{ i18nText-siteTitle }}</title>
+		<meta content="width=device-width, initial-scale=1" name="viewport">
+		{% include metadata.html %}
+		{% include resources-inc/head.html %}
+	</head>
+	<body class="home page-type-nav{% if page.pageclass %} {{ page.pageclass }}{% endif %}" vocab="http://schema.org/" typeof="WebPage">
+		{% include sites/inc-skiplinks.html %}
+		{% include header/header.html %}
```
Comment (GormFrank) [inline templates/home/layouts/home.html:21]: The home page must not have a breadcrumb.

**Review by GormFrank — CHANGES_REQUESTED** [review]
@Ricokola super travail :)

Tant qu'à modifier les colonne de header, aussi bien par le fait même améliorer la colonne pour le sign in button en mobile; c'est-à-dire de changer `col-xs-offset-7 col-xs-5` par `col-xs-offset-6 col-xs-6` (ne pas oublié d'ajouter cette référence dans le CSS pour garder le margin-top négatif et garder l'ancienne référence aussi pour ne pas causer de major). C'est seulement pour ne pas que le bouton tombe sur deux lignes sur téléphone mobile.

**Review by duboisp — COMMENTED** [review]
* [ ] Review and local testing

pre-approved at the technical review.

File: sites/header/header-docs-fr.html [hunk 8]
```diff
@@ -0,0 +1,350 @@
...
+		<li><a href="deprecated/header-auth-v2.html" hreflang="en">Page de contenu incluant l'en-tête version 2.0 ainsi que la section d'authentification</a></li>
+		<li><a href="deprecated/home-auth-v2.html" hreflang="en">Page d'accueil incluant l'en-tête version 2.0 ainsi que la section d'authentification</a></li>
+		<li><a href="deprecated/home-v2.html" hreflang="en">Page d'accueil incluant l'en-tête version 2.0</a></li>
+	</ul>
+</details>
+
+<ul>
+	<li><a href="deprecated/header-v1.html" hreflang="en">En-tête Version 1.0</a></li>
+	<li><a href="../../templates/home/deprecated/home-v1.html" hreflang="en">Page d'accueil En-tête Version 1.0</a></li>
+</ul>
```
Comment (duboisp) [inline sites/header/header-docs-fr.html:350]: Please put those link in a "details/summary" element like you did in English.

**Review by duboisp — CHANGES_REQUESTED** [review]
* [ ] Keep all deprecated "home" template at the same place, within the template home and not with the sites header. (4 files to moves)

---

Question:
Only for the home page - The working example for the version 3 of the structural don't match the changes described in the documentation. Is that intentional or not? I know that at one point our home page didn't followed the update on our generic pages.

**Comment by duboisp** [comment]
pre-approved upon successful local testing + code review

File: sites/header/header-docs-fr.html [hunk 9]
```diff
@@ -0,0 +1,390 @@
...
+		</tr>
+		<tr>
+			<th>Version 1.0</th>
+			<td>n.a.</td>
+			<td>Version 1.0</td>
+			<td>Version 1.0</td>
+			<td>n.a.</td>
+		</tr>
+	</table>
+</details>
```
Comment (duboisp) [inline sites/header/header-docs-fr.html:92]: A traduire ou ajouter le language tag et une note disant que quelques partie de ce document nécessite une traduction.

**Review by duboisp — CHANGES_REQUESTED** [review]
I review and tested. There is one little additional change and the rest look good.

* [ ] Fail SC 3.1.2 Language of part for the "header-docs-fr.html"

**Review by duboisp — APPROVED** [review]

**Comment by duboisp** [comment]
@Ricokola or @GormFrank could one of you resolve the merge conflict. Thanks

FYI - The conflict files are:
* sites/authentication/authentication-en.html
* sites/authentication/authentication-fr.html

Pre-approved once the conflict is resolve and the change is only about the above file

**Comment by GormFrank** [comment]
@duboisp Rebased!

**Review by duboisp — APPROVED** [review]
Change requested was made and I reviewed and tested the last modified file since my last review and it works as expected.

**Pull request merged**